libs/gundeck-types/src/Gundeck/Types/BulkPush.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+
+-- | This module defines the types used by the Cannon API.  It is contained in package gundeck-types
+-- for the pragmatic reason that it allows us to re-use types from the gundeck API.  (This move can
+-- be justified by picturing Cannon as a local service that makes only sense in the context of a
+-- Gundeck.)
+module Gundeck.Types.BulkPush where
+
+import Data.Aeson
+import Data.Id
+import GHC.Generics
+import Gundeck.Types.Notification
+
+
+data PushTarget = PushTarget
+    { ptUserId :: !UserId
+    , ptConnId :: !ConnId
+    } deriving ( Eq, Ord, Show
+               , Generic
+               )
+
+instance FromJSON PushTarget where
+    parseJSON = withObject "push target object" $ \hm ->
+        PushTarget <$> (hm .: "userId") <*> (hm .: "connId")
+
```

Comment:
[nit-pick]: Can we use `underscores` instead `userId`, `connId` and `notifId` plz?

Reply:
fb084f003a658584c086d09f68be00bcb0f19cfb